src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13109,5 +13161,512 @@ public static void Main()
+        [Theory]
+        [InlineData("sbyte")]
+        [InlineData("byte")]
+        [InlineData("short")]
+        [InlineData("ushort")]
+        [InlineData("int")]
+        [InlineData("uint")]
+        [InlineData("long")]
+        [InlineData("ulong")]
+        [InlineData("nint")]
+        [InlineData("nuint")]
+        [InlineData("float")]
+        [InlineData("double")]
+        [InlineData("decimal")]
+        [InlineData("E", "E.FIELD", "FIELD")]
+        [InlineData("bool", "true", "True")]
+        [InlineData("char", "'a'", "a")]
+        [InlineData("string", @"""a string""", "a string")]
+        [InlineData("C", "null", "")]
+        public void LambdaDefaultParameter_AllConstantValueTypes(string parameterType, string defaultValue = "0", string expectedOutput = "0")
+        {
+            var source = $@"
+using System;
+public class Program
+{{
+    public enum E 
+    {{
+        FIELD
+    }}
+    
+    class C {{}}
+
+    public static void Main()
+    {{
+        var lam = ({parameterType} p = {defaultValue}) => p;
+        Console.WriteLine(lam());
+    }}
+}}
+";
+            CompileAndVerify(source, expectedOutput: expectedOutput);
+        }
+
+        [Fact]
+        public void LambdaDefaultParameter_TargetTypedValidLiteralConversion()
+        {
+            TestDiagnosticsInMain(
+"""
+var lam = (short s = 1) => { };
+""");
+        }
+
+        [Fact]
+        public void LambdaDefaultParameter_TargetTypeInvalidLiteralConversion()
+        {
+            TestDiagnosticsInMain(
+"""
+var lam = (short s = 32768) => { };
+""", expectedDiagnostics:
+                 // (6,26): error CS1750: A value of type 'int' cannot be used as a default parameter because there are no standard conversions to type 'short'
+                 //         var lam = (short s = 32768) => { };
+                 Diagnostic(ErrorCode.ERR_NoConversionForDefaultParam, "s").WithArguments("int", "short").WithLocation(6, 26));
+        }
+
+        [ConditionalFact(typeof(NoIOperationValidation))]
+        public void LambdaDefaultParameter_TargetTypedValidNonLiteralConversion()
+        {
+            TestDiagnosticsInMain(
+"""
+const float floatConst = 1f;
+var lam = (double d = floatConst) => { };
+""");
+        }
+
+        [ConditionalFact(typeof(NoIOperationValidation))]
+        public void LambdaDefaultParameter_InterpolatedStringHandler()
+        {
+            var source = """
+using System.Runtime.CompilerServices;
+using System;
+using System.Text;
+
+[InterpolatedStringHandler]
+public ref struct CustomInterpolatedStringHandler
+{
+    StringBuilder builder;
+
+    public CustomInterpolatedStringHandler(int literalLength, int formattedCount) { builder = new StringBuilder(literalLength); }
+    public void AppendLiteral(string s) { builder.Append(s + "<<CustomInterpolatedStringHandler>>"); }
+    public void AppendFormatted<T>(T t) { builder.Append(t?.ToString()); }
+    internal string GetFormattedText() => builder.ToString();
+}
```

Comment:
Rather than manually defining a handler, you can use `GetInterpolatedStringCustomHandlerType`. https://github.com/dotnet/roslyn/blob/main/src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs#L2621

Reply:
Great, thanks. Used: b57f1c40ef7cfefb6eaaea19005c7b513d48ea98